### PR TITLE
fix links autocomplete suggestions: empty post title fallback

### LINKED ISSUE #3573: Links autocomplete suggestions: posts with no title should have a fallback title
When inserting a link and the links suggestions appears, posts with no title don't have any text, just an empty button. Visually, there's nothing that looks clickable. Assistive technologies will announce just "button".

To reproduce:
- create a new post with no title, make sure to insert a paragraph block with some text you can search for later
- publish
- edit another post, select some text, and click the Link button in the toolbar
- search for some text inserted in the content of the post with no title

List of suggestions (notice the space between two of the suggestions is the empty button of the post with no title):

screen shot 2017-11-20 at 14 37 48

Use the keyboard arrow keys to navigate through and highlight the suggestions: 

screen shot 2017-11-20 at 14 25 12

Worth noting this doesn't work well on the 4.9 classic editor too.

screen shot 2017-11-20 at 14 38 35

Worth also reminding in other places in the admin WordPress uses the `(no title)` fallback:

screen shot 2017-11-20 at 14 42 45

### PR DESCRIPTION
**Pull request by vladanost**
## Description

Fixes https://github.com/WordPress/gutenberg/issues/3573

## How Has This Been Tested?

Manually tested in the browser.

## Types of changes

Added fallback title for post with no title

### CODE REVIEW

File: blocks/url-input/index.js [hunk 1]
```diff
@@ -215,7 +215,7 @@ class UrlInput extends Component {
 								onClick={ () => this.props.onChange( post.link ) }
 								aria-selected={ index === selectedSuggestion }
 							>
-								{ post.title.rendered }
+								{ post.title.rendered || `( ${ __( 'no title' ) } )` }
```
Comment (youknowriad) [inline blocks/url-input/index.js]: Maybe a bit simpler like this: `post.title.rendered || __( 'no title' )`

Comment (vladanost) [reply blocks/url-input/index.js]: Sure, I can do that. I added the brackets so they are more in line with how no title posts are displayed in other places in WP admin as @afercia noted.

Comment (youknowriad) [reply blocks/url-input/index.js]: Oh sorry for the confusion, I see what you mean now and you're right we should add the parenthesis. `post.title.rendered || __( '(no title)' )` 

A quick search in the WP source code shows that the parenthesis are included in the translated message.

**Review by youknowriad — APPROVED** [review]
Thanks for the fix

**Pull request merged**